### PR TITLE
Officially Create Version 0.1.0

### PR DESCRIPTION
**Pull request by AndrewGaspar**
## PR Summary

This change primarily sets the version number in CMakeList.txt and puts some boiler plate about the initial release in CHANGELOG.md.

Also changes the rules a little about tagged releases - prefixes the releases with "v", and also calls for branches that track the latest minor release for each major version.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
I thought we were going to be including a date in the version... or am I confused by something else?

**Comment by AndrewGaspar** [comment]
That was originally discussed, but we decided to use semantic versioning instead, modified to treat the minor version as the major version for now until we reach 1.0.

**Comment by AndrewGaspar** [comment]
Semantic versioning: https://semver.org/

**Review by JoshuaSBrown — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,26 +2,14 @@
 
 ## Current develop
 
-### Added (new features/APIs/variables/...)
-- Introduced CHANGELOG.md
-
-### Changed (changing behavior/API/variables/...)
```
Comment (pgrete) [inline CHANGELOG.md:8]: Was the 4 way skeleton removed on purpose or by accident?
I'm in favor of keeping it 4 way.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,7 +28,8 @@ We aim at creating a new release everyone 4 to 6 weeks.
 The decision on creating a new release is made during the bi-weekly calls.
 A release consists of of merging `develop` into `stable` and create a new tag for that version
 using a modified [semantic versioning](https://semver.org/) scheme.
-Releases will be tagged `0.MAJOR.MINOR` given the current rapid development.
+Releases will be tagged `v0.MAJOR.MINOR` given the current rapid development.
+Release branches named `v0.MAJOR` will track the latest minor release.
```
Comment (pgrete) [inline CONTRIBUTING.md]: Would this result in having a persistent release branch for every major release?
I'm wondering what the use case is compared to directly using the tags, e.g., if I want to update the Parthenon submodule in a downstream code I'd have to pull the changes and commit the updated version to the downstream repo anyway.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,26 +2,14 @@
...
-
-
-## Release 0.MAJOR.MINOR
-Note:
-- will only contain the subsections of develop that were filled
-- will (ideally) include the PR that introduced the change
-
 ### Added (new features/APIs/variables/...)
 - lorem (#11)
 - ipsum (#14)
```
Comment (Yurlungur) [inline CHANGELOG.md]: I get that these are examples... but should they stay?

**Review by Yurlungur — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (AndrewGaspar) [reply CHANGELOG.md:8]: It was an accident. I'll fix.

File: CONTRIBUTING.md (hunk 2 above)
Comment (AndrewGaspar) [reply CONTRIBUTING.md]: Yeah, I suppose. I think it just makes it easier to get the latest ABI compatible version if you want to update from an older 0.1 to a newer 0.1, without having to know what the latest minor release is.

Comment (pgrete) [reply CONTRIBUTING.md]: Given the current rapid change I suggest that we hold off the separate branch until the first official 1.x release.
For now we could add a hint to the docs saying that with a simple `git tag --list 'v0.MAJOR.*'` one can find the latest minor release.

**Comment by AndrewGaspar** [comment]
@pgrete Should be good to go now.

**Review by pgrete — APPROVED** [review]

**Comment by pgrete** [comment]
Shall we add info from #214 (and #259) to the changelog here? I think this would be quite important as it actually changes the API.

**Comment by AndrewGaspar** [comment]
I'll add info from #214. #259 doesn't seem to have any user-facing issues - just additional documentation.

**Pull request merged**